### PR TITLE
btcmarkets: update order submission

### PR DESCRIPTION
**Pull request by shazbert**
# PR Description

* Adds formatting methods.
* Change target amount to quote amount to make it a little bit more obvious and add commentary.
* Remove price and post only from json request when not needed. 

Fixes # (issue)

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and
also consider improving test coverage whilst working on a certain feature or package.

- [ ] go test ./... -race
- [ ] golangci-lint run
- [ ] Test X

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally and on Github Actions/AppVeyor with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: exchanges/btcmarkets/btcmarkets.go [hunk 1]
```diff
@@ -314,13 +318,57 @@ func (b *BTCMarkets) GetTradeByID(ctx context.Context, id string) (TradeHistoryD
...
+// getTimeInForce returns a string depending on the options in order.Submit
+func (b *BTCMarkets) getTimeInForce(s *order.Submit) string {
+	if s.ImmediateOrCancel {
+		return "IOC"
+	}
+	if s.FillOrKill {
+		return "FOK"
+	}
+	return "" // GTC (good till cancelled, default value
+}
```
Comment (gloriousCode) [inline exchanges/btcmarkets/btcmarkets.go:362]: Please add coverage  to these two🌟

File: exchanges/btcmarkets/btcmarkets_test.go [hunk 2]
```diff
@@ -899,3 +899,55 @@ func TestTrim(t *testing.T) {
 		})
 	}
 }
+
+func TestFormatOrderType(t *testing.T) {
+	_, err := b.formatOrderType(order.Type("SWOOON"))
```
Comment (gloriousCode) [inline exchanges/btcmarkets/btcmarkets_test.go:904]: Please add `t.Parallel()`

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Nice! Just some micro nits 🛩️

File: exchanges/btcmarkets/btcmarkets_test.go [hunk 3]
```diff
@@ -202,26 +202,48 @@ func TestGetTradeByID(t *testing.T) {
 	}
 }
 
-func TestNewOrder(t *testing.T) {
+func TestSubmitOrder(t *testing.T) {
```
Comment (gloriousCode) [inline exchanges/btcmarkets/btcmarkets_test.go:205]: To be a huge jerk, I appreciate this test, however this does remove the test on just the API function. Could you please re-add the test and keep this one?

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Thanks for making those changes! Just one thing

**Review by gloriousCode — APPROVED** [review]
tACKeroo! Thanks for making those changes

**Review by thrasher- — APPROVED** [review]

**Pull request merged**